tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import re
+import tempfile
+from contextlib import contextmanager
+from typing import Optional
+
+import click
+
+from tdp.cli.queries import get_planned_deployment_log
+from tdp.cli.session import get_session
+from tdp.cli.utils import collections, database_dsn
+from tdp.core.models.deployment_log import DeploymentLog
+
+logger = logging.getLogger("tdp").getChild("edit")
+
+
+def _get_header_message(deployment_log_id: int, temp_file_name: str) -> str:
+    """Returns the header message to be displayed in the temporary file."""
+    return f"""\
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+# Modify this file to edit the deployment plan.
+# Lines beginning with '#' are comments and will be ignored.
+#
+# Current plan id: {deployment_log_id}
+# Temporary file: {temp_file_name}
+#
+# Execution Order:
+# - Operations are executed in the order they appear below.
+# - You can add, remove, or rearrange operations.
+# - Use `tdp nodes` command to view a list of available operations.
+#
+# Operations options:
+# - Specify a host with the `on` keyword.
+#   Example: hdfs_datanode_config on worker_02
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
+
+
+def _parse_line(line: str) -> tuple[str, Optional[str], Optional[list[str]]]:
+    """Parses a line which contains an operation, and eventually a host and extra vars.
+
+    Args:
+        line: line to be parsed
```

Comment:
To respect consistency, styling of the line must be like this:
`line: Line to be parsed.`

Reply:
Done, I also added a '.'.